app/uk/gov/hmrc/exports/repositories/UpdateSubmissionsTransactionalOps.scala

```diff
@@ -147,37 +150,23 @@ class UpdateSubmissionsTransactionalOps @Inject() (
     latestEnhancedStatus: EnhancedStatus
   ): Future[Option[Submission]] = (for {
     decId <- action.decId
-    update <- submissionStatus match {
-      case REJECTED | CUSTOMS_POSITION_DENIED =>
-        Some(
-          Json.obj(
-            "$set" -> Json.obj(
-              "latestDecId" -> decId,
-              "latestVersionNo" -> action.versionNo,
-              "mrn" -> mrn,
-              "latestEnhancedStatus" -> ON_HOLD,
-              "enhancedStatusLastUpdated" -> summary.dateTimeIssued,
-              "actions" -> actions
-            )
-          )
-        )
-      case CUSTOMS_POSITION_GRANTED =>
-        Some(
-          Json.obj(
-            "$set" -> Json.obj(
-              "latestDecId" -> decId,
-              "latestVersionNo" -> action.versionNo,
-              "mrn" -> mrn,
-              "latestEnhancedStatus" -> latestEnhancedStatus,
-              "enhancedStatusLastUpdated" -> summary.dateTimeIssued,
-              "actions" -> actions
-            )
-          )
-        )
-      case _ => None
+    latestEnhancedStatus <- submissionStatus match {
```

Comment:
Nit: this syntax is confusing me. Would you mind to move the `match` expression to a proper def/val? Something like...
```
val enhancedStatus = submissionStatus match {
      case REJECTED | CUSTOMS_POSITION_DENIED => Some(ON_HOLD)
      case CUSTOMS_POSITION_GRANTED => Some(latestEnhancedStatus)
      case _  => None
}
for {
  ...
```